src/figuresClasses.ts

```diff
@@ -1,11 +1,93 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+type Color = 'red' | 'green' | 'blue';
 
-export class Circle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    function isRequiredSides(
+      side1: number,
+      side2: number,
+      side3: number,
+    ): boolean {
+      const minSide = Math.min(side1, side2, side3);
+      const maxSide = Math.max(side1, side2, side3);
+      const mediumSide = side1 + side2 + side3 - maxSide - minSide;
+
+      if (
+        side1 <= 0 ||
+        side2 <= 0 ||
+        side3 <= 0 ||
+        maxSide >= minSide + mediumSide
+      ) {
+        return false;
+      }
+
+      return true;
+    }
+
+    if (!isRequiredSides(this.a, this.b, this.c)) {
+      throw new Error('error sides');
```

Comment:
[CODE STYLE] - If you throw an error try to provide its detailed description in error message. Instead of 'error sides', provide a message that explains what is wrong with the sides, for example: 'Triangle sides must satisfy the triangle inequality theorem'.